refactor: update `blas/ext/base/gcusumkbn2` to follow current project conventions

headlessNode opened the pull request:
## Description

> What is the purpose of this pull request?

This pull request:

-   update the package to follow current project conventions.

## Related Issues

> Does this pull request have any related issues?

No.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

headlessNode committed:
refactor: update to follow current project conventions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

headlessNode committed:
docs: apply code review suggestion

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

headlessNode committed:
docs: apply code review suggestion

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

headlessNode committed:
Merge branch 'stdlib-js:develop' into refactor-gcusumkbn2

kgryte reviewed (changes requested):
Looks like this PR needs similar to changes to https://github.com/stdlib-js/stdlib/pull/4412. Would you be willing to update, @headlessNode?

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/README.md:
```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, offset parameters support indexing semantics based on starting indices. For example, to calculate the cumulative sum of every other element in the strided input array starting from the second element and to store in the last `N` elements of the strided output array starting from the last element:
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/README.md:
```suggestion
var x = discreteUniform( 10, -100, 100, {
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/docs/repl.txt:
```suggestion
    buffer, the offset parameters support indexing semantics based on starting
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/examples/index.js:
```suggestion
var x = discreteUniform( 10.0, -100, 100, {
```
```suggestion
var x = discreteUniform( 10, -100, 100, {
```

headlessNode committed:
fix: apply review suggestions

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
fix: apply review suggestions

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
fix: apply review suggestions

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
fix: apply review suggestions

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
fix: apply review suggestion

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.js:
@headlessNode Would you mind refactoring back to generic arrays?

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.ndarray.js:
Same comment.

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.js:
```suggestion
	'dtype': 'generic'
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.ndarray.js:
```suggestion
	'dtype': 'generic'
```

headlessNode committed:
Update lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.js

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
Update lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.ndarray.js

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode commented:
@kgryte changes applied.

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.js:
```suggestion
var zeros = require( '@stdlib/array/zeros' );
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.js:
```suggestion
	var y = zeros( len, options.dtype );
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.js:
```suggestion
var zeros = require( '@stdlib/array/zeros' );
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.js:
```suggestion
	var y = zeros( len, options.dtype );
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.ndarray.js:
```suggestion
var zeros = require( '@stdlib/array/zeros' );
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.ndarray.js:
```suggestion
var zeros = require( '@stdlib/array/zeros' );
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.ndarray.js:
```suggestion
	var y = zeros( len, options.dtype );
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/benchmark/benchmark.ndarray.js:
```suggestion
	var y = zeros( len, options.dtype );
```

headlessNode committed:
refactor: apply review suggestion

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

kgryte reviewed (approved):
LGTM
